docs/src/configuration.md

```diff
@@ -170,3 +170,22 @@ WIP
 ]
 ```
 ````
+
+## `.gitignore`
+
+### Retaining files when step is reset
+
+Reseting step will remove all untracked files from the project working directory (`<PROJECT_DASHED_NAME>`). `.gitignore` files can be used to prevent git from removing certain files when step is reset. Normal gitignore patterns applies.
+
+````admonish info
+For defining pattern for whole course:
+```txt
+<COURSE_DIR>/
+|-- .gitignore
+```
+For specific project:
+```txt
+|-- <PROJECT_DASHED_NAME>/
+|   |-- .gitignore
+```
+````
```

Comment:
I do not think an example is necessary.

```suggestion
```

Reply:
I agree it's a bit much elaborate. I was trying to put slight emphasis that there are two possible locations for the file.